app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddSidewalkSurface.kt

```diff
@@ -0,0 +1,147 @@
+package de.westnordost.streetcomplete.quests.surface
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.ANYTHING_UNPAVED
+import de.westnordost.streetcomplete.data.meta.SIDEWALK_SURFACE_KEYS
+import de.westnordost.streetcomplete.data.meta.hasCheckDateForKey
+import de.westnordost.streetcomplete.data.meta.removeCheckDatesForKey
+import de.westnordost.streetcomplete.data.meta.updateCheckDateForKey
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.OUTDOORS
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.PEDESTRIAN
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.WHEELCHAIR
+
+class AddSidewalkSurface : OsmFilterQuestType<SidewalkSurfaceAnswer>() {
+
+    override val elementFilter = """
+        ways with
+            highway ~ trunk|trunk_link|primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential
+            and area != yes
+            and motorroad != yes
+            and (
+                sidewalk = both or sidewalk = left or sidewalk = right or
+                (sidewalk:left = yes and sidewalk:right ~ yes|no|separate) or
+                (sidewalk:right = yes and sidewalk:left ~ yes|no|separate)
+            )
+            and (
+                ${SIDEWALK_SURFACE_KEYS.joinToString(" and ") {"!$it"}}
+                or sidewalk:both:surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and sidewalk:both:surface older today -4 years
+                or sidewalk:left:surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and sidewalk:left:surface older today -4 years
+                or sidewalk:right:surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and sidewalk:right:surface older today -4 years
```

Comment:
`sidewalk:right:surface older today -4 years`

looks for `check_date:sidewalk:right:surface`. The cycleway quest adds only `check_date:cycleway`, it would be consistent if this quest also did this unless there is a good reason not to.
We don't want this quest to add three different check_dates.

Also, given this filter is already so complex, maybe we could just remove these lines. Just check every 8 years always.